# Week 3 Homework

In [1]:
import pandas as pd 
import numpy as np

## Task 1. Import dataset as FIFA.

In [2]:
df = pd.read_csv("fifa19.csv")
df.head()

Unnamed: 0      ID               Name  Age  \
0           0  158023           L. Messi   31   
1           1   20801  Cristiano Ronaldo   33   
2           2  190871          Neymar Jr   26   
3           3  193080             De Gea   27   
4           4  192985       K. De Bruyne   27   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   

                  Club  ... Composure Marking StandingTackle  SlidingTackle  \
0         FC Barcelona  ...      96.0    33.0           28.0           26.0   
1             Juventus  ...      95.0    28.0           31.0           23.0   
2  Paris Saint-Germain  ...      94.0    27.0           24.0           33.0   
3    Manchester United  ...      68.0    15.0           21.0           13.0   
4      Manchester City  ...      88.0    68.0           58.0           51.0   

  GKDiving  GKHandling  GKKicking  GKPositioning GKReflexes Release Clause  
0      6.0        11.0       15.0           14.0        8.0        €226.5M  
1      7.0        11.0       15.0           14.0       11.0        €127.1M  
2      9.0         9.0       15.0           15.0       11.0        €228.1M  
3     90.0        85.0       87.0           88.0       94.0        €138.6M  
4     15.0        13.0        5.0           10.0       13.0        €196.4M  

[5 rows x 89 columns]

## Task 2. Show first 12 entries of dataset.

In [3]:
df.head(12)

Unnamed: 0      ID               Name  Age  \
0            0  158023           L. Messi   31   
1            1   20801  Cristiano Ronaldo   33   
2            2  190871          Neymar Jr   26   
3            3  193080             De Gea   27   
4            4  192985       K. De Bruyne   27   
5            5  183277          E. Hazard   27   
6            6  177003          L. Modrić   32   
7            7  176580          L. Suárez   31   
8            8  155862       Sergio Ramos   32   
9            9  200389           J. Oblak   25   
10          10  188545     R. Lewandowski   29   
11          11  182521           T. Kroos   28   

                                             Photo Nationality  \
0   https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1    https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2   https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3   https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4   https://cdn.sofifa.org/players/4/19/192985.png     Belgium   
5   https://cdn.sofifa.org/players/4/19/183277.png     Belgium   
6   https://cdn.sofifa.org/players/4/19/177003.png     Croatia   
7   https://cdn.sofifa.org/players/4/19/176580.png     Uruguay   
8   https://cdn.sofifa.org/players/4/19/155862.png       Spain   
9   https://cdn.sofifa.org/players/4/19/200389.png    Slovenia   
10  https://cdn.sofifa.org/players/4/19/188545.png      Poland   
11  https://cdn.sofifa.org/players/4/19/182521.png     Germany   

                                   Flag  Overall  Potential  \
0   https://cdn.sofifa.org/flags/52.png       94         94   
1   https://cdn.sofifa.org/flags/38.png       94         94   
2   https://cdn.sofifa.org/flags/54.png       92         93   
3   https://cdn.sofifa.org/flags/45.png       91         93   
4    https://cdn.sofifa.org/flags/7.png       91         92   
5    https://cdn.sofifa.org/flags/7.png       91         91   
6   https://cdn.sofifa.org/flags/10.png       91         91   
7   https://cdn.sofifa.org/flags/60.png       91         91   
8   https://cdn.sofifa.org/flags/45.png       91         91   
9   https://cdn.sofifa.org/flags/44.png       90         93   
10  https://cdn.sofifa.org/flags/37.png       90         90   
11  https://cdn.sofifa.org/flags/21.png       90         90   

                   Club  ... Composure Marking StandingTackle  SlidingTackle  \
0          FC Barcelona  ...      96.0    33.0           28.0           26.0   
1              Juventus  ...      95.0    28.0           31.0           23.0   
2   Paris Saint-Germain  ...      94.0    27.0           24.0           33.0   
3     Manchester United  ...      68.0    15.0           21.0           13.0   
4       Manchester City  ...      88.0    68.0           58.0           51.0   
5               Chelsea  ...      91.0    34.0           27.0           22.0   
6           Real Madrid  ...      84.0    60.0           76.0           73.0   
7          FC Barcelona  ...      85.0    62.0           45.0           38.0   
8           Real Madrid  ...      82.0    87.0           92.0           91.0   
9       Atlético Madrid  ...      70.0    27.0           12.0           18.0   
10    FC Bayern München  ...      86.0    34.0           42.0           19.0   
11          Real Madrid  ...      85.0    72.0           79.0           69.0   

   GKDiving  GKHandling  GKKicking  GKPositioning GKReflexes Release Clause  
0       6.0        11.0       15.0           14.0        8.0        €226.5M  
1       7.0        11.0       15.0           14.0       11.0        €127.1M  
2       9.0         9.0       15.0           15.0       11.0        €228.1M  
3      90.0        85.0       87.0           88.0       94.0        €138.6M  
4      15.0        13.0        5.0           10.0       13.0        €196.4M  
5      11.0        12.0        6.0            8.0        8.0        €172.1M  
6      13.0         9.0        7.0           14.0        9.0        €137.4M  
7      27

## Task 3. Get information about dataset.

In [6]:
df.describe()

Unnamed: 0             ID           Age       Overall     Potential  \
count  18207.000000   18207.000000  18207.000000  18207.000000  18207.000000   
mean    9103.000000  214298.338606     25.122206     66.238699     71.307299   
std     5256.052511   29965.244204      4.669943      6.908930      6.136496   
min        0.000000      16.000000     16.000000     46.000000     48.000000   
25%     4551.500000  200315.500000     21.000000     62.000000     67.000000   
50%     9103.000000  221759.000000     25.000000     66.000000     71.000000   
75%    13654.500000  236529.500000     28.000000     71.000000     75.000000   
max    18206.000000  246620.000000     45.000000     94.000000     95.000000   

            Special  International Reputation     Weak Foot   Skill Moves  \
count  18207.000000              18159.000000  18159.000000  18159.000000   
mean    1597.809908                  1.113222      2.947299      2.361308   
std      272.586016                  0.394031      0.660456      0.756164   
min      731.000000                  1.000000      1.000000      1.000000   
25%     1457.000000                  1.000000      3.000000      2.000000   
50%     1635.000000                  1.000000      3.000000      2.000000   
75%     1787.000000                  1.000000      3.000000      3.000000   
max     2346.000000                  5.000000      5.000000      5.000000   

       Jersey Number  ...     Penalties     Composure       Marking  \
count   18147.000000  ...  18159.000000  18159.000000  18159.000000   
mean       19.546096  ...     48.548598     58.648274     47.281623   
std        15.947765  ...     15.704053     11.436133     19.904397   
min         1.000000  ...      5.000000      3.000000      3.000000   
25%         8.000000  ...     39.000000     51.000000     30.000000   
50%        17.000000  ...     49.000000     60.000000     53.000000   
75%        26.000000  ...     60.000000     67.000000     64.000000   
max        99.000000  ...     92.000000     96.000000     94.000000   

       StandingTackle  SlidingTackle      GKDiving    GKHandling  \
count    18159.000000   18159.000000  18159.000000  18159.000000   
mean        47.697836      45.661435     16.616223     16.391596   
std         21.664004      21.289135     17.695349     16.906900   
min          2.000000       3.000000      1.000000      1.000000   
25%         27.000000      24.000000      8.000000      8.000000   
50%         55.000000      52.000000     11.000000     11.000000   
75%         66.000000      64.000000     14.000000     14.000000   
max         93.000000      91.000000     90.000000     92.000000   

          GKKicking  GKPositioning    GKReflexes  
count  18159.000000   18159.000000  18159.000000  
mean      16.232061      16.388898     16.710887  
std       16.502864      17.034669     17.955119  
min        1.000000       1.000000      1.000000  
25%        8.000000       8.000000      8.000000  
50%       11.000000      11.000000     11.000000  
75%       14.000000      14.000000     14.000000  
max       91.000000      90.000000     94.000000  

[8 rows x 44 columns]

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18207 entries, 0 to 18206
Data columns (total 89 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                18207 non-null  int64  
 1   ID                        18207 non-null  int64  
 2   Name                      18207 non-null  object 
 3   Age                       18207 non-null  int64  
 4   Photo                     18207 non-null  object 
 5   Nationality               18207 non-null  object 
 6   Flag                      18207 non-null  object 
 7   Overall                   18207 non-null  int64  
 8   Potential                 18207 non-null  int64  
 9   Club                      17966 non-null  object 
 10  Club Logo                 18207 non-null  object 
 11  Value                     18207 non-null  object 
 12  Wage                      18207 non-null  object 
 13  Special                   18207 non-null  int64  
 14  Prefer

## Task 4. Show the number of observations in dataset

In [9]:
df.shape

(18207, 89)

we have 18207 rows (observations) and 89 columns (features)

## Task 5. Print column names

In [10]:
df.columns

Index(['Unnamed: 0', 'ID', 'Name', 'Age', 'Photo', 'Nationality', 'Flag',
       'Overall', 'Potential', 'Club', 'Club Logo', 'Value', 'Wage', 'Special',
       'Preferred Foot', 'International Reputation', 'Weak Foot',
       'Skill Moves', 'Work Rate', 'Body Type', 'Real Face', 'Position',
       'Jersey Number', 'Joined', 'Loaned From', 'Contract Valid Until',
       'Height', 'Weight', 'LS', 'ST', 'RS', 'LW', 'LF', 'CF', 'RF', 'RW',
       'LAM', 'CAM', 'RAM', 'LM', 'LCM', 'CM', 'RCM', 'RM', 'LWB', 'LDM',
       'CDM', 'RDM', 'RWB', 'LB', 'LCB', 'CB', 'RCB', 'RB', 'Crossing',
       'Finishing', 'HeadingAccuracy', 'ShortPassing', 'Volleys', 'Dribbling',
       'Curve', 'FKAccuracy', 'LongPassing', 'BallControl', 'Acceleration',
       'SprintSpeed', 'Agility', 'Reactions', 'Balance', 'ShotPower',
       'Jumping', 'Stamina', 'Strength', 'LongShots', 'Aggression',
       'Interceptions', 'Positioning', 'Vision', 'Penalties', 'Composure',
       'Marking', 'StandingTackle', 'SlidingT

## Task 6. Show 5000th observation.

In [16]:
df.iloc[4999,:]

Unnamed: 0                                                  4999
ID                                                        202050
Name                                                    T. Bruns
Age                                                           26
Photo             https://cdn.sofifa.org/players/4/19/202050.png
                                       ...                      
GKHandling                                                  11.0
GKKicking                                                   13.0
GKPositioning                                               14.0
GKReflexes                                                   9.0
Release Clause                                             €3.4M
Name: 4999, Length: 89, dtype: object

## Task 7. Select user_id column.

In [18]:
df.loc[:,"ID"]

0        158023
1         20801
2        190871
3        193080
4        192985
          ...  
18202    238813
18203    243165
18204    241638
18205    246268
18206    246269
Name: ID, Length: 18207, dtype: int64

## Task 8. Sort teams by Club and Name columns.

In [19]:
df.sort_values(by = ["Club","Name"])

Unnamed: 0      ID          Name  Age  \
7657         7657  238594        A. Dej   26   
11244       11244  244280       A. Fein   19   
7184         7184  213644      A. Geipl   26   
6917         6917  238666    A. Nandzik   25   
12363       12363  238667     A. Odabas   24   
...           ...     ...           ...  ...   
9373         9373  246406    X. Arreaga   23   
3037         3037  207797     Y. Banana   26   
1602         1602  216335  Y. Gazinskiy   28   
3710         3710  245665   Y. Ouattara   34   
4667         4667  223208     Z. Kalmár   23   

                                                Photo  Nationality  \
7657   https://cdn.sofifa.org/players/4/19/238594.png       Poland   
11244  https://cdn.sofifa.org/players/4/19/244280.png      Germany   
7184   https://cdn.sofifa.org/players/4/19/213644.png      Germany   
6917   https://cdn.sofifa.org/players/4/19/238666.png      Germany   
12363  https://cdn.sofifa.org/players/4/19/238667.png      Germany   
...                                               ...          ...   
9373   https://cdn.sofifa.org/players/4/19/246406.png      Ecuador   
3037   https://cdn.sofifa.org/players/4/19/207797.png     Cameroon   
1602   https://cdn.sofifa.org/players/4/19/216335.png       Russia   
3710   https://cdn.sofifa.org/players/4/19/245665.png  Ivory Coast   
4667   https://cdn.sofifa.org/players/4/19/223208.png      Hungary   

                                       Flag  Overall  Potential  \
7657    https://cdn.sofifa.org/flags/37.png       67         70   
11244   https://cdn.sofifa.org/flags/21.png       64         79   
7184    https://cdn.sofifa.org/flags/21.png       68         70   
6917    https://cdn.sofifa.org/flags/21.png       68         71   
12363   https://cdn.sofifa.org/flags/21.png       63         69   
...                                     ...      ...        ...   
9373    https://cdn.sofifa.org/flags/57.png       66         71   
3037   https://cdn.sofifa.org/flags/103.png       73         75   
1602    https://cdn.sofifa.org/flags/40.png       75         75   
3710   https://cdn.sofifa.org/flags/108.png       72         72   
4667    https://cdn.sofifa.org/flags/23.png       71         78   

                       Club  ... Composure Marking StandingTackle  \
7657    SSV Jahn Regensburg  ...      63.0    65.0           58.0   
11244   SSV Jahn Regensburg  ...      57.0    49.0           63.0   
7184    SSV Jahn Regensburg  ...      58.0    68.0           71.0   
6917    SSV Jahn Regensburg  ...      61.0    65.0           64.0   
12363   SSV Jahn Regensburg  ...      55.0    64.0           62.0   
...                     ...  ...       ...     ...            ...   
9373                    NaN  ...      44.0    69.0           74.0   
3037                    NaN  ...      57.0    77.0           76.0   
1602                    NaN  ...      74.0    73.0           74.0   
3710                    NaN  ...      72.0    17.0           28.0   
4667                    NaN  ...      70.0    57.0           45.0   

       SlidingTackle GKDiving  GKHandling  GKKicking  GKPositioning  \
7657            51.0     10.0        11.0       14.0           10.0   
11244           64.0     10.0         8.0       12.0           12.0   
7184            72.0     12.0        14.0        8.0           12.0   
6917            64.0      9.0        11.0        6.0           14.0   
12363           58.0     12.0         9.0        6.0            7.0   
...              ...      ...         ...        ...            ...   
9373            71.0      7.0        13.0        8.0           11.0   
3037            69.0     14.0         7.0        7.0            7.0   
1602            71.0     13.0        15.0       11.0           15.0   
3710            21.0      8.0        15.0       15.0           13.0   
4667            43.0     14.0         7.0       12.0           15.0   

      GKReflexes Release Clause  
7657        13.0          €1.5M  
11244        6.0            NaN  


## Task 9. Filter Players whose Overall score is more than 90.

In [22]:
df[df["Overall"] > 90]

Unnamed: 0      ID               Name  Age  \
0           0  158023           L. Messi   31   
1           1   20801  Cristiano Ronaldo   33   
2           2  190871          Neymar Jr   26   
3           3  193080             De Gea   27   
4           4  192985       K. De Bruyne   27   
5           5  183277          E. Hazard   27   
6           6  177003          L. Modrić   32   
7           7  176580          L. Suárez   31   
8           8  155862       Sergio Ramos   32   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   
2  https://cdn.sofifa.org/players/4/19/190871.png      Brazil   
3  https://cdn.sofifa.org/players/4/19/193080.png       Spain   
4  https://cdn.sofifa.org/players/4/19/192985.png     Belgium   
5  https://cdn.sofifa.org/players/4/19/183277.png     Belgium   
6  https://cdn.sofifa.org/players/4/19/177003.png     Croatia   
7  https://cdn.sofifa.org/players/4/19/176580.png     Uruguay   
8  https://cdn.sofifa.org/players/4/19/155862.png       Spain   

                                  Flag  Overall  Potential  \
0  https://cdn.sofifa.org/flags/52.png       94         94   
1  https://cdn.sofifa.org/flags/38.png       94         94   
2  https://cdn.sofifa.org/flags/54.png       92         93   
3  https://cdn.sofifa.org/flags/45.png       91         93   
4   https://cdn.sofifa.org/flags/7.png       91         92   
5   https://cdn.sofifa.org/flags/7.png       91         91   
6  https://cdn.sofifa.org/flags/10.png       91         91   
7  https://cdn.sofifa.org/flags/60.png       91         91   
8  https://cdn.sofifa.org/flags/45.png       91         91   

                  Club  ... Composure Marking StandingTackle  SlidingTackle  \
0         FC Barcelona  ...      96.0    33.0           28.0           26.0   
1             Juventus  ...      95.0    28.0           31.0           23.0   
2  Paris Saint-Germain  ...      94.0    27.0           24.0           33.0   
3    Manchester United  ...      68.0    15.0           21.0           13.0   
4      Manchester City  ...      88.0    68.0           58.0           51.0   
5              Chelsea  ...      91.0    34.0           27.0           22.0   
6          Real Madrid  ...      84.0    60.0           76.0           73.0   
7         FC Barcelona  ...      85.0    62.0           45.0           38.0   
8          Real Madrid  ...      82.0    87.0           92.0           91.0   

  GKDiving  GKHandling  GKKicking  GKPositioning GKReflexes Release Clause  
0      6.0        11.0       15.0           14.0        8.0        €226.5M  
1      7.0        11.0       15.0           14.0       11.0        €127.1M  
2      9.0         9.0       15.0           15.0       11.0        €228.1M  
3     90.0        85.0       87.0           88.0       94.0        €138.6M  
4     15.0        13.0        5.0           10.0       13.0        €196.4M  
5     11.0        12.0        6.0            8.0        8.0        €172.1M  
6     13.0         9.0        7.0           14.0        9.0        €137.4M  
7     27.0        25.0       31.0           33.0       37.0          €164M  
8     11.0         8.0        9.0            7.0       11.0        €104.6M  

[9 rows x 89 columns]

## Task 10. Remove 'Unnamed:0' column.

In [23]:
df.drop('Unnamed: 0', axis=1,inplace = True)

In [25]:
df.head(2)

ID               Name  Age  \
0  158023           L. Messi   31   
1   20801  Cristiano Ronaldo   33   

                                            Photo Nationality  \
0  https://cdn.sofifa.org/players/4/19/158023.png   Argentina   
1   https://cdn.sofifa.org/players/4/19/20801.png    Portugal   

                                  Flag  Overall  Potential          Club  \
0  https://cdn.sofifa.org/flags/52.png       94         94  FC Barcelona   
1  https://cdn.sofifa.org/flags/38.png       94         94      Juventus   

                                      Club Logo  ... Composure Marking  \
0  https://cdn.sofifa.org/teams/2/light/241.png  ...      96.0    33.0   
1   https://cdn.sofifa.org/teams/2/light/45.png  ...      95.0    28.0   

   StandingTackle SlidingTackle  GKDiving  GKHandling  GKKicking  \
0            28.0          26.0       6.0        11.0       15.0   
1            31.0          23.0       7.0        11.0       15.0   

  GKPositioning GKReflexes Release Clause  
0          14.0        8.0        €226.5M  
1          14.0       11.0        €127.1M  

[2 rows x 88 columns]

## Task 11. How many distinct Positions exist in dataset?

In [30]:
len(df['Position'].unique())

28

## Task 12. Show null values

In [59]:
df.isnull().sum()

ID                   0
Name                 0
Age                  0
Photo                0
Nationality          0
                  ... 
GKHandling          48
GKKicking           48
GKPositioning       48
GKReflexes          48
Release Clause    1564
Length: 88, dtype: int64

We can see that the output is truncated. Therefore, I want to see only the rows which has null values, and without truncation. So, I will convert the series into dictionary. 

In [54]:
null_values = df.isnull().sum()
null_values = null_values[null_values > 0]
null_values = null_values.to_frame()

In [63]:
null_values.to_dict()

{0: {'Club': 241,
  'Preferred Foot': 48,
  'International Reputation': 48,
  'Weak Foot': 48,
  'Skill Moves': 48,
  'Work Rate': 48,
  'Body Type': 48,
  'Real Face': 48,
  'Position': 60,
  'Jersey Number': 60,
  'Joined': 1553,
  'Loaned From': 16943,
  'Contract Valid Until': 289,
  'Height': 48,
  'Weight': 48,
  'LS': 2085,
  'ST': 2085,
  'RS': 2085,
  'LW': 2085,
  'LF': 2085,
  'CF': 2085,
  'RF': 2085,
  'RW': 2085,
  'LAM': 2085,
  'CAM': 2085,
  'RAM': 2085,
  'LM': 2085,
  'LCM': 2085,
  'CM': 2085,
  'RCM': 2085,
  'RM': 2085,
  'LWB': 2085,
  'LDM': 2085,
  'CDM': 2085,
  'RDM': 2085,
  'RWB': 2085,
  'LB': 2085,
  'LCB': 2085,
  'CB': 2085,
  'RCB': 2085,
  'RB': 2085,
  'Crossing': 48,
  'Finishing': 48,
  'HeadingAccuracy': 48,
  'ShortPassing': 48,
  'Volleys': 48,
  'Dribbling': 48,
  'Curve': 48,
  'FKAccuracy': 48,
  'LongPassing': 48,
  'BallControl': 48,
  'Acceleration': 48,
  'SprintSpeed': 48,
  'Agility': 48,
  'Reactions': 48,
  'Balance': 48,
  'ShotPower

In this way, we can observe all rows as **dictionary keys**, and the number of null values in the corresponding rows as **dictionary values**

## Task 13. Fill empty values in 'Release Clause' column with 'unknown'

In [65]:
# before filling how many nulls are there in the column: 
df['Release Clause'].isnull().sum()

np.int64(0)

In [62]:
df['Release Clause'].fillna('unknown', axis = 0, inplace = True)

In [64]:
# after filling, the number of nulls: 
df['Release Clause'].isnull().sum()

np.int64(0)

## Task 14. Find mean of Age.

In [70]:
df['Age'].mean()

np.float64(25.122205745043114)

## Task 15. Find maximum of Shot Power.

In [72]:
df['ShotPower'].max()

np.float64(95.0)

## Task 16. Show footballer has maximum Shot Power.

In [75]:
df[df['ShotPower'] == df['ShotPower'].max()]

ID               Name  Age  \
1  20801  Cristiano Ronaldo   33   

                                           Photo Nationality  \
1  https://cdn.sofifa.org/players/4/19/20801.png    Portugal   

                                  Flag  Overall  Potential      Club  \
1  https://cdn.sofifa.org/flags/38.png       94         94  Juventus   

                                     Club Logo  ... Composure Marking  \
1  https://cdn.sofifa.org/teams/2/light/45.png  ...      95.0    28.0   

   StandingTackle SlidingTackle  GKDiving  GKHandling  GKKicking  \
1            31.0          23.0       7.0        11.0       15.0   

  GKPositioning GKReflexes Release Clause  
1          14.0       11.0        €127.1M  

[1 rows x 88 columns]

## Task 17. Describe Data 

In [80]:
df.describe()

ID           Age       Overall     Potential       Special  \
count   18207.000000  18207.000000  18207.000000  18207.000000  18207.000000   
mean   214298.338606     25.122206     66.238699     71.307299   1597.809908   
std     29965.244204      4.669943      6.908930      6.136496    272.586016   
min        16.000000     16.000000     46.000000     48.000000    731.000000   
25%    200315.500000     21.000000     62.000000     67.000000   1457.000000   
50%    221759.000000     25.000000     66.000000     71.000000   1635.000000   
75%    236529.500000     28.000000     71.000000     75.000000   1787.000000   
max    246620.000000     45.000000     94.000000     95.000000   2346.000000   

       International Reputation     Weak Foot   Skill Moves  Jersey Number  \
count              18159.000000  18159.000000  18159.000000   18147.000000   
mean                   1.113222      2.947299      2.361308      19.546096   
std                    0.394031      0.660456      0.756164      15.947765   
min                    1.000000      1.000000      1.000000       1.000000   
25%                    1.000000      3.000000      2.000000       8.000000   
50%                    1.000000      3.000000      2.000000      17.000000   
75%                    1.000000      3.000000      3.000000      26.000000   
max                    5.000000      5.000000      5.000000      99.000000   

           Crossing  ...     Penalties     Composure       Marking  \
count  18159.000000  ...  18159.000000  18159.000000  18159.000000   
mean      49.734181  ...     48.548598     58.648274     47.281623   
std       18.364524  ...     15.704053     11.436133     19.904397   
min        5.000000  ...      5.000000      3.000000      3.000000   
25%       38.000000  ...     39.000000     51.000000     30.000000   
50%       54.000000  ...     49.000000     60.000000     53.000000   
75%       64.000000  ...     60.000000     67.000000     64.000000   
max       93.000000  ...     92.000000     96.000000     94.000000   

       StandingTackle  SlidingTackle      GKDiving    GKHandling  \
count    18159.000000   18159.000000  18159.000000  18159.000000   
mean        47.697836      45.661435     16.616223     16.391596   
std         21.664004      21.289135     17.695349     16.906900   
min          2.000000       3.000000      1.000000      1.000000   
25%         27.000000      24.000000      8.000000      8.000000   
50%         55.000000      52.000000     11.000000     11.000000   
75%         66.000000      64.000000     14.000000     14.000000   
max         93.000000      91.000000     90.000000     92.000000   

          GKKicking  GKPositioning    GKReflexes  
count  18159.000000   18159.000000  18159.000000  
mean      16.232061      16.388898     16.710887  
std       16.502864      17.034669     17.955119  
min        1.000000       1.000000      1.000000  
25%        8.000000       8.000000      8.000000  
50%       11.000000      11.000000     11.000000  
75%       14.000000      14.000000     14.000000  
max       91.000000      90.000000     94.000000  

[8 rows x 43 columns]